controllers/subnamespace_controller.go

```diff
@@ -119,30 +149,37 @@ func GetNewChildNamespace(ownerNamespace v1.Namespace, subspace danav1alpha1.Sub
 	}
 }
 
-func GetChildNamespaceName(ownerNamespace v1.Namespace, subspace danav1alpha1.Subnamespace) string {
+func GetChildNamespaceName(ownerNamespace *v1.Namespace, subspace *danav1alpha1.Subnamespace) string {
 	return ownerNamespace.Name + "-" + subspace.Name
 }
-func (r *SubnamespaceReconciler) CreateNamespace(ctx context.Context, childNamespace v1.Namespace, ownerNamespace v1.Namespace, subspace danav1alpha1.Subnamespace) (requeueResult bool, err error) {
-	childNamespace = GetNewChildNamespace(ownerNamespace, subspace)
-	if err := r.Create(ctx, &childNamespace); err != nil {
-		return false, err
-	} else {
-		return true, nil
+func (r *SubnamespaceReconciler) CreateNamespace(ctx context.Context, log logr.Logger, ownerNamespace *v1.Namespace, subspace *danav1alpha1.Subnamespace) error {
+	var childNamespace = GetNewChildNamespace(ownerNamespace, subspace)
+	if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &childNamespace, func() error {
+		return nil
+	}); err != nil {
+		log.Error(err, "Could not create child namespace")
+		return err
 	}
+	return nil
+
 }
 
-func (r *SubnamespaceReconciler) InitializeSubspace(ctx context.Context, ownerNamespace v1.Namespace, subspace danav1alpha1.Subnamespace, childNamespaceName string) error {
-	if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &subspace, func() error {
+func (r *SubnamespaceReconciler) InitSubspace(ctx context.Context, log logr.Logger, ownerNamespace *v1.Namespace, subspace *danav1alpha1.Subnamespace, childNamespaceName string) error {
+	if _, err := ctrl.CreateOrUpdate(ctx, r.Client, subspace, func() error {
 		subspace.SetAnnotations(map[string]string{
 			danav1alpha1.Pointer: childNamespaceName,
 		})
 		subspace.Status.Phase = danav1alpha1.Missing
-		if err := ctrl.SetControllerReference(&ownerNamespace, &subspace, r.Scheme); err != nil {
+		if err := ctrl.SetControllerReference(ownerNamespace, subspace, r.Scheme); err != nil {
 			return err
 		}
 		return nil
 	}); err != nil {
-		return err
+		if !apierrors.IsConflict(err) {
+			log.Error(err, "Could not update Subspace Phase from 'None' to 'Missing'")
+			return err
+		}
+		log.Info("newer resource version exists")
```

Comment:
`apierrors.IsConflict` indicates that you are trying to create/update a resource that already had a newer version. We want to ignore this error since we will reconcile on the newer version in the future